### ISSUE TITLE
Reduce the redundant dimension of gradient for LieGroup optimization

### ISSUE THREAD

**Issue by MarkChenYutian** [pypose/pypose#387]
## Summary

Currently, when optimizing a LieGroup parameter using `pp.optim`, the gradient's shape is always aligned with the parameter's shape by padding zeros at the end due to the following way of handling parameter updates.

https://github.com/pypose/pypose/blob/33cb63a9c42a6d29aa168e913d3e7cdd597a60ca/pypose/optim/optimizer.py#L96-L101

These zeros are then discarded in the corresponding `ltype.add_` method (e.g. the `SE3Type.add_` below dropped the 7-th col of gradient since they are always zero):

https://github.com/pypose/pypose/blob/33cb63a9c42a6d29aa168e913d3e7cdd597a60ca/pypose/lietensor/lietensor.py#L443-L444

While this design maintains the compatibility with torch's first order optimizers, the padded zeros significantly degrades the numeric condition of second-order optimizers. (Since the zero-padded jacobians $J$ are used to estimate hessian by $H\approx J^TJ$, lead to singular matrix in the solver)

This padded zero can be removed by using the following logic in second-order optimizer base class `_Optimizer`:

```python
    def update_parameter(self, params, step):
        """
        params will be updated by calling this function
        """
        def get_gradient_size(p: torch.Tensor | pp.LieTensor) -> torch.Size:
            if isinstance(p, pp.LieTensor):
                return p.shape[:-1] + p.ltype.manifold
            else:
                return p.shape
        
        grad_params = [p for p in params if p.requires_grad]
        grad_shapes = [get_gradient_size(p) for p in grad_params]
        steps       = torch.split(step, [s.numel() for s in grad_shapes])
        for p, d, s in zip(grad_params, grad_shapes, steps):
            p.add_(s.view(d))

```

## Improvements

Experiment shows that removing the padding zeros in the jacobian significantly improve the numeric stability of estimated hessian matrix

Below is the numeric stability analysis for backend optimizer of a visual odometry for the `A` matrix on this line:

https://github.com/pypose/pypose/blob/33cb63a9c42a6d29aa168e913d3e7cdd597a60ca/pypose/optim/optimizer.py#L284

Current implementation:

<img width="396" height="71" alt="Image" src="https://github.com/user-attachments/assets/73da208f-6268-4273-b7fc-64ddfa92782b" />

Removing the redundant zero cols:

<img width="499" height="91" alt="Image" src="https://github.com/user-attachments/assets/d84492c6-363e-4c7f-a1af-9e3fbf078aa8" />

The condition number improved significantly (average condition number from the sampling period reduced from $1\times 10^{14}$ to $1\times 10^{5}$). Meaning that the solution is significantly more stable.

## Risks

(Please list features that can break because of this refactoring and how you intend to solve that.)

## Involved components

(List files or directories that will be changed by the refactoring.)

## Optional: Intended side effects

(If the refactoring involves changes apart from the main improvements, list them here.
It may be a good idea to create separate issues and link them here.)

## Optional: Missing test coverage

(
If you are aware of tests that need to be written or adjusted apart from unit tests for the changed components,
please list them here.
)
